### PR TITLE
Fix for a rcon timing out

### PR DESCRIPTION
**Pull request by aapokaapo**
This will fix the command sent via rcon to be sent until there is a return from the server. No more timing out

### CODE REVIEW

**Comment by aapokaapo** [comment]
This has to be reviewed ASAP. DPLib will break at some point without this.

File: dplib/server.py [hunk 1]
```diff
@@ -502,11 +502,33 @@ def rcon(self, command):
...
+        ret = None
+        while not ret:
+            # start with a socket at 5-second timeout
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.settimeout(5.0)
+            try:
+                sock.connect((self.__hostname, self.__port))
+            except socket.error:
+                print('Socket connect failed! Trying again')
+                traceback.print_exc()
```
Comment (aapokaapo) [inline dplib/server.py]: this is a useless line

**Comment by mRokita** [comment]
Damn, nested while loops are quite ugly.
You are right that UDP packets can be lost though, and this issue needs a fix (on localhost it's no big teal though)

**Comment by mRokita** [comment]
I will write a fix from scratch someday.

**Comment by aapokaapo** [comment]
Good to hear. I am inexperienced with sockets, so I was merely poking around there. This works for me, but a cleaner code would certainly be better. Contact me if you want any help somewhere!

**Comment by aapokaapo** [comment]
Having this fixed is crucial tho. If there is one UDP packet lost, the script stops running. I was running my scripts on VPS and on the long run the scripts would crash whenever the server was under a bit of stress. So this is an important stability fix

**Comment by mRokita** [comment]
Can't you handle it in your code?

**Pull request closed without merging**